Tried js approach on converting class to css

pr0h0 opened the pull request:
Should I continue working on this?

It needs to be written as class and add more stuff, but it's just beginning.

pr0h0 committed:
Tried js approach on converting class to css

radubrehar commented on uniss.js:
```diff
@@ -0,0 +1,107 @@
+    document.head.appendChild(ourStyleElement);
+  }
+
+  const searchQuery = `[class*='${options.prefix}-']`; // search string for elemnt selection
```
we should definitely not check and do querySelector based on the `un` prefix.

to save bandwidth, a better approach would be to generate all css at runtime and inject it into the document. Doing this only for elements that match query selector, would mean elements inserted later would not be matched

radubrehar replied:
but first it's worth knowing the size of the whole css, after minification and gzip is applied.

pr0h0 replied:
> we should definitely not check and do querySelector based on the un prefix.

we shouldn't use querySelector at all or ?
we just get elements with prefix in className and then filter those classes to make sure is valid 

> to save bandwidth, a better approach would be to generate all css at runtime and inject it into the document. Doing this only for elements that match query selector, would mean elements inserted later would not be matched

Static properties like display, position.. we can generate as static CSS file, and other properties like width, height... we can generate on runtime, at page load and at each change of class.

We can use [MutationObserver ](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) for looking at those changes

radubrehar commented:
@pr0h0 see my comments

pr0h0 committed:
rewrote as class, added caching

pr0h0 commented:
Rewrote it as class. 
Added watchers so on every child add or class change, it will handle it and generate new classes.
If any, added caching and check for existing classes so we don't have save rules more than once.
Tested with generating 1000 elements and toggling 10 classes on each, 10k ops, took 300-500ms on my laptop.

pr0h0 commented:
@radubrehar can i get any feedback on this?

pr0h0 committed:
update git ignore
